### ISSUE TITLE
Any way to disable globalTypes generation?

### ISSUE THREAD

**Issue by tushar-singh** [apollographql/apollo-tooling#1179]
I'm trying to use codegen in an existing codebase and can't run it on all files as that fails due to numerous issues (imported fragments from other modules, anonymous operations, etc)

What I've instead been doing is running codegen on new code. It works but whenever globalTypes is generated it overwrites previous data – this leaves older operation types without enums

Is there a way to disable globalTypes generation and keep enums local?

**Comment by dsebastien** [comment]
I'm also interested by this, but for different reasons: I'm generating code for a few GraphQL queries using the Apollo CLI codegen feature, but in my case the globalTypes.ts file is empty, so I assume that I simply don't need it.

The issue is that I'm using the Create React App scripts in my project, and it enforces the `isolatedModules` option to `true`. Having empty files is not allowed when that option is enabled.

For now my workaround is to instruct the CLI to generate the globalTypes file outside the compilation scope, but this is a dirt hack that I'd like to avoid :)

Any idea of another (cleaner) solution? I feel like disabling the generation of that empty file would be ideal in my case.

**Comment by ansoncen** [comment]
I simply exclude the `globalTypes.ts` in my `tsconfig.json` and it workes as promised.

**Comment by macrozone** [comment]
i think apollo should not generate the global-types if its empty. it results in errors also for next.js because that forces isolatedModules to be true as well

**Comment by wonjerry** [comment]
+1

**Comment by macrozone** [comment]
damn, just run into that again. Does anyone know a workaround?

**Comment by Manubi** [comment]
any news on this one? as excluding it in my tsconfig.json doesn't seem to work for me.

**Comment by natterstefan** [comment]
Hi, I am also curious if there's a fix for that.

FTR: there seems to be an open PR related to this, but it's still open: https://github.com/apollographql/apollo-tooling/pull/2163.

**Comment by ryan-rushton** [comment]
I got around this issue with the setting `--globalTypesFile src/__generated-globals__/globalTypes.d.ts`.

**Comment by Manubi** [comment]
> I got around this issue with the setting `--globalTypesFile src/__generated-globals__/globalTypes.d.ts`.

You are putting it outside of the compilation scope which works put is not really "clean". A simple `--globalTypesFile false` would be enough put somehow no one seems to care...

**Comment by ryan-rushton** [comment]
@Manubi I totally agree, just adding another workaround for people that need to get on with it in the mean time.

**Comment by mhmoritz** [comment]
If `globalTypes.ts` is empty and you often generate types with the help of a custom script in `package.json` like I do, you can simply delete the empty file programatically:
```json
"types:generate": "npx apollo codegen:generate --localSchemaFile=graphql-schema.json --target=typescript --outputFlat --addTypename src/types && rm src/types/globalTypes.ts"

```

**Comment by macrozone** [comment]
any workaround for this? Its a really annoying problem that would be easy to avoid.

**Comment by pamplonapaulo** [comment]
> > I got around this issue with the setting `--globalTypesFile src/__generated-globals__/globalTypes.d.ts`.
> 
> You are putting it outside of the compilation scope which works put is not really "clean". A simple `--globalTypesFile false` would be enough but somehow no one seems to care...

I actually care and your solution helped me. Thank you very much @Manubi !

**Comment by toxeus** [comment]
FYI: the type definition generator [has been deprecated](https://github.com/apollographql/apollo-tooling/commit/573ff2be902d8aaf15a1e6b6eec5202c35803e05#diff-598fae964b3b6081f4f7970c254bb0c3bfaf8e66cc76d113832f4b9ec5d5a2e9R135).

https://github.com/apollographql/apollo-tooling/blob/apollo%402.34.0/packages/apollo/src/commands/client/codegen.ts#L135-L138

Hence, I don't believe issues like this one will be fixed anymore.
